webapp/components/multiselect/multiselect_list.jsx

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {cmdOrCtrlPressed} from 'utils/utils.jsx';
+import Constants from 'utils/constants.jsx';
+const KeyCodes = Constants.KeyCodes;
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+export default class MultiSelectList extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.defaultOptionRenderer = this.defaultOptionRenderer.bind(this);
+        this.handleArrowPress = this.handleArrowPress.bind(this);
+        this.setSelected = this.setSelected.bind(this);
+
+        this.toSelect = -1;
+
+        this.state = {
+            selected: -1
+        };
+    }
+
+    componentDidMount() {
+        document.addEventListener('keydown', this.handleArrowPress);
```

Comment:
Do these need to be global? Could they just be on a div wrapping the entire contents of the list?